### PR TITLE
Replaced JAVA_BIN to corresponding Test_JDK_HOME in Test

### PR DESCRIPTION
**Pull request by LongyuZhang**
- Replaced JAVA_BIN to corresponding TEST_JDK_HOME to avoid ambiguity in test
- makefile, build_test, and settings.mk will be changed later along with JAVA_BIN in OpenJDK.
- Merged codes which were used to handle different jdk versions but not needed now
- Split codes which need to handle different jdk version due to removal of JAVA_BIN
[ci skip]

Signed-off-by: Longyu <longyu.zhang@ibm.com>

### CODE REVIEW

**Comment by LongyuZhang** [comment]
reviewers: @llxia @renfeiw

**Comment by LongyuZhang** [comment]
Tested internally with both JDK8 and JDK11 on:
sanity.functional
extended.functional
extended.system
openjdk.functional (on JDK8 get same result as nightly build: EXECUTED: 9   PASSED: 7   FAILED: 2)

File: test/TestConfig/scripts/build_test.xml [hunk 1]
```diff
@@ -95,38 +87,4 @@
 
 		<chmod file="${ascii.test.dir}/*.mk" perm="755"/>
 	</target>
-
-
-	<target name="check_sdk">
```
Comment (llxia) [inline test/TestConfig/scripts/build_test.xml]: We should keep `check_sdk ` for now as it is still in use.

**Comment by llxia** [comment]
Could we try run thrstatetest, vmLifecyleTests, SharedClassesSysVTesting with `TEST_JDK_HOME/bin` for JDK8? Thanks

**Comment by LongyuZhang** [comment]
> Could we try run thrstatetest, vmLifecyleTests, SharedClassesSysVTesting with `TEST_JDK_HOME/bin` for JDK8? Thanks

SharedClassesSysVTesting has merged all jdk versions into one test, and works fine;
 thrstatetest and vmLifecyleTests failed after merging all jdk version, because it looks for 'j9vm' folder which is under jre/lib for jdk8;
Same tests for different versions have been changed to use the same test name as suggested.

File: test/TestConfig/scripts/build_test.xml (hunk 1 above)
Comment (LongyuZhang) [reply test/TestConfig/scripts/build_test.xml]: check_sdk has been reserved for the uses of other repos.

File: test/functional/VM_Test/playlist.xml [hunk 2]
```diff
@@ -102,6 +102,11 @@
 		<groups>
 			<group>functional</group>
 		</groups>
+        <subsets>
+			<subset>8</subset>
```
Comment (llxia) [inline test/functional/VM_Test/playlist.xml]: Could we remove the extra space?

File: test/functional/cmdLineTests/gptest/playlist.xml [hunk 3]
```diff
@@ -51,5 +51,41 @@
 		<types>
 			<type>native</type>
 		</types>
+        <subsets>
+            <subset>8</subset>
+        </subsets>
```
Comment (llxia) [inline test/functional/cmdLineTests/gptest/playlist.xml]: Could we make the `<subsets>` alined? Same as below.

File: test/functional/VM_Test/playlist.xml (hunk 2 above)
Comment (LongyuZhang) [reply test/functional/VM_Test/playlist.xml]: Alinements have been fixed. Thanks.

File: test/functional/cmdLineTests/gptest/playlist.xml (hunk 3 above)
Comment (LongyuZhang) [reply test/functional/cmdLineTests/gptest/playlist.xml]: Alinements have been fixed. Thanks.

**Comment by llxia** [comment]
jenkins test sanity all jdk11

**Comment by llxia** [comment]
jenkins test all all jdk8

**Comment by llxia** [comment]
Jenkins test sanity,extended all jdk11

**Comment by llxia** [comment]
Jenkins test sanity,extended all jdk8

**Comment by llxia** [comment]
Jenkins test sanity,extended all jdk8

**Comment by LongyuZhang** [comment]
JDK8 extended.functional_s390x_linux test failure is related to #5449 
JDK8 extended.functional_x86-64_linux_xl test failure is related to #5489

**Comment by llxia** [comment]
Thanks @LongyuZhang

**Comment by pshipton** [comment]
This seems to be causing a lot of UnsatisfiedLinkError problems and failure to run tests that use native binaries. I'm reverting this while we figure it out.

**Pull request merged**